### PR TITLE
obj: Add Btf::to_bytes

### PR DESCRIPTION
**Pull request by dave-tucker**
This allows for parsed BTF to be re-encoded such that it could be loaded
in to the kernel

Signed-off-by: Dave Tucker <dave@dtucker.co.uk>

### CODE REVIEW

File: aya/src/obj/btf/btf.rs [hunk 1]
```diff
@@ -291,6 +293,26 @@ impl Btf {
             type_id: root_type_id,
         })
     }
+
+    #[allow(dead_code)]
```
Comment (alessandrod) [inline aya/src/obj/btf/btf.rs:297]: 🤔

**Review by alessandrod — CHANGES_REQUESTED** [review]
I think you should use this instead of all those from_raw_parts: https://github.com/aya-rs/aya/blob/352e54b72405b5e9f21a947ff0146f3ba162b78a/aya/src/obj/mod.rs#L638

Maybe put it as `pub(crate)` in utils.rs or something?

File: aya/src/obj/btf/btf.rs (hunk 1 above)
Comment (dave-tucker) [reply aya/src/obj/btf/btf.rs]: oops, fixed it by making it pub, which was what I intended to do :laughing:

**Comment by dave-tucker** [comment]
ugh. going to close this as it doesn't make sense on it's own as the code that uses it isn't here..

**Pull request closed without merging**